common/functions/src/scalars/expressions/cast.rs

```diff
@@ -199,6 +214,13 @@ fn string_to_datetime(date_str: impl AsRef<[u8]>) -> Option<NaiveDateTime> {
     s.and_then(|c| NaiveDateTime::parse_from_str(c, "%Y-%m-%d %H:%M:%S").ok())
 }
 
+// TODO support timezone
+#[inline]
+fn string_to_datetime64(date_str: impl AsRef<[u8]>) -> Option<NaiveDateTime> {
```

Comment:
Is there any stateless test about datetime64 feature?

https://github.com/datafuselabs/databend/blob/4af0e109e8fba997ed78ad9e3f38b0cb02015f0b/tests/suites/0_stateless/02_0012_function_datetimes.sql

Reply:
I have added some tests, PTAL @sundy-li